RHEL7编译出错

samuel@localhost:~/src/libae$ make
cc -c -std=c99 -g -O0 -Wno-parentheses -Wno-switch-enum -Wno-unused-value src/anet.c -o src/anet.o
src/anet.c: In function ‘anetGenericResolve’:
src/anet.c:178:21: error: storage size of ‘hints’ isn’t known
     struct addrinfo hints, *info;
                     ^
src/anet.c:182:48: error: ‘AI_NUMERICHOST’ undeclared (first use in this function)
     if (flags & ANET_IP_ONLY) hints.ai_flags = AI_NUMERICHOST;

--- a/COPYING
+++ b/COPYING
@@ -1,4 +1,4 @@
-Copyright (c) 2006-2014, Salvatore Sanfilippo
+Copyright (c) 2006-2015, Salvatore Sanfilippo
 All rights reserved.
 
 Redistribution and use in source and binary forms, with or without modification, are permitted provided that the following conditions are met:

--- a/Makefile
+++ b/Makefile
@@ -1,6 +1,6 @@
 SRC = src/ae.c src/anet.c
 OBJ = ${SRC:.c=.o}
-CFLAGS = -std=c99 -g -O0 -Wno-parentheses -Wno-switch-enum -Wno-unused-value
+CFLAGS = -Wno-parentheses -Wno-switch-enum -Wno-unused-value
 
 libae.a: $(OBJ)
 	$(AR) -rc $@ $(OBJ)

--- a/Makefile
+++ b/Makefile
@@ -8,10 +8,10 @@ libae.a: $(OBJ)
 %.o: %.c
 	$(CC) -c $(CFLAGS) $< -o $@
 
-timer: libae.a example/timer.o
+timer: example/timer.o libae.a
 	$(CC) $^ -o $@
 
-echo: libae.a example/echo.o
+echo: example/echo.o libae.a
 	$(CC) $^ -o $@
 
 clean:

--- a/src/ae.c
+++ b/src/ae.c
@@ -404,7 +404,7 @@ int aeProcessEvents(aeEventLoop *eventLoop, int flags)
             int fd = eventLoop->fired[j].fd;
             int rfired = 0;
 
-	    /* note the fe->mask & mask & ... code: maybe an already processed
+            /* note the fe->mask & mask & ... code: maybe an already processed
              * event removed an element that fired and we still didn't
              * processed, so we check if the event is still valid. */
             if (fe->mask & mask & AE_READABLE) {

--- a/src/fmacros.h
+++ b/src/fmacros.h
@@ -1,5 +1,61 @@
+/*
+ * Copyright (c) 2009-2012, Salvatore Sanfilippo <antirez at gmail dot com>
+ * All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions are met:
+ *
+ *   * Redistributions of source code must retain the above copyright notice,
+ *     this list of conditions and the following disclaimer.
+ *   * Redistributions in binary form must reproduce the above copyright
+ *     notice, this list of conditions and the following disclaimer in the
+ *     documentation and/or other materials provided with the distribution.
+ *   * Neither the name of Redis nor the names of its contributors may be used
+ *     to endorse or promote products derived from this software without
+ *     specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+ * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+ * ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR CONTRIBUTORS BE
+ * LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+ * CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+ * SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+ * INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+ * CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ * POSSIBILITY OF SUCH DAMAGE.
+ */
+
 #ifndef _REDIS_FMACRO_H
 #define _REDIS_FMACRO_H
 
+#define _BSD_SOURCE
+
+#if defined(__linux__)
+#define _GNU_SOURCE
+#define _DEFAULT_SOURCE
+#endif
+
+#if defined(_AIX)
+#define _ALL_SOURCE
+#endif
+
+#if defined(__linux__) || defined(__OpenBSD__)
+#define _XOPEN_SOURCE 700
+/*
+ * On NetBSD, _XOPEN_SOURCE undefines _NETBSD_SOURCE and
+ * thus hides inet_aton etc.
+ */
+#elif !defined(__NetBSD__)
+#define _XOPEN_SOURCE
+#endif
+
+#if defined(__sun)
+#define _POSIX_C_SOURCE 199506L
+#endif
+
+#define _LARGEFILE_SOURCE
+#define _FILE_OFFSET_BITS 64
 
 #endif

